common/tests/parse_config_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include <atomic>
+#include <fstream>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "common/thrift_router.h"
+
+static const char* g_config_v1 =
+  "{"
+  "  \"a\": {"
+  "  \"num_leaf_segments\": 1,"
+  "  \"127.0.0.1:8090\": [\"00000\"]"
+  "   },"
+  "  \"b\": {"
+  "  \"num_leaf_segments\": 1,"
+  "  \"127.0.0.1:8090\": [\"00000\"]"
+  "   }"
+  "}";
+
+
+// This test requires ASAN enabled to function correctly
```

Comment:
Is the comments true? i.e. will the test pass without ASAN, and when we run with cmake test. 
iiuc, you meant the test is meant to cature the UseAfterFree issue which is only surfaced with ASAN. In that case, I think it would be nice to add some comments on what's the expected output and some instructions to get it running.